src/main/java/labs/introtoprogramming/lab5/object/Parser/StringUtils.java

```diff
@@ -0,0 +1,287 @@
+            if (count >= listLength) {
+                break;
+            }
+            itemStart = count;
+            itemEnd = itemStart;
+            while (itemEnd < listLength) {
+                if ((listChars[itemEnd] != ' ') && (listChars[itemEnd] != '\n') && (listChars[itemEnd] != '\t')) {
+                    itemEnd++;
+                } else {
+                    break;
+                }
+            }
+            itemLength = itemEnd - itemStart;
+            returnList.add(Integer.parseInt(new String(listChars, itemStart, itemLength)));
+
+            count = itemEnd;
+        }
+
+        int returnArray[] = new int[returnList.size()];
+        for (int loopi = 0; loopi < returnList.size(); loopi++) {
+            returnArray[loopi] = returnList.get(loopi);
+        }
+        return returnArray;
+    }
+
+    public static int[] parseListVertexNTuples(String list, int expectedValuesPerTuple) {
+        if (list == null) {
+            return null;
+        }
+        if (list.equals("")) {
+            return null;
+        }
+        String[] vertexStrings = parseWhitespaceList(list);
+
+
+        ArrayList<Integer> returnList = new ArrayList<Integer>();
+        Integer emptyMarker = new Integer(BuilderInterface.EMPTY_VERTEX_VALUE);
+
+        for (int loopi = 0; loopi < vertexStrings.length; loopi++) {
+
+            parseVerticeNTuple(vertexStrings[loopi], returnList, emptyMarker, expectedValuesPerTuple);
+        }
+
+        int returnArray[] = new int[returnList.size()];
+        for (int loopi = 0; loopi < returnList.size(); loopi++) {
+            returnArray[loopi] = returnList.get(loopi);
+        }
+        return returnArray;
+    }
+
+    private static void parseVerticeNTuple(String list, ArrayList<Integer> returnList, Integer emptyMarker, int expectedValueCount) {
+
+
+
+        String[] numbers = parseList('/', list);
+
+        int foundCount = 0;
+
+        int index = 0;
+        while (index < numbers.length) {
+
+            if (numbers[index].trim().equals("")) {
+
+                returnList.add(emptyMarker);
+            } else {
+
+                returnList.add(Integer.parseInt(numbers[index]));
+            }
+            foundCount++;
+            index++;
+        }
+        while (foundCount < expectedValueCount) {
+            returnList.add(emptyMarker);
+            foundCount++;
+        }
+    }
+
+    public static String[] parseList(char delim, String list) {
+        if (list == null) {
+            return null;
+        }
+        if (list.equals("")) {
+            return null;
+        }
+
+        ArrayList<String> returnVec = new ArrayList<String>();
+        String[] returnArray = null;
+
+
+        char listChars[];
+        listChars = new char[list.length()];
+        list.getChars(0, list.length(), listChars, 0);
+
+        int count = 0;
+        int itemStart = 0;
+        int itemEnd = 0;
+        String newItem = null;
+
+        while (count < listChars.length) {
```

Comment:
Why don't you use split? And you can try to use stream to write it more compactly.

Reply:
Done)